File: modeles/pharmacie.js



const { DataTypes } = require("sequelize");
const sequelize = require('../configuration/sequelizeconfig');

const pharmacie = sequelize.define('pharmacie', {
    id_pharmacie: {
      type: DataTypes.INTEGER,
      primaryKey: true,
      autoIncrement: true,
    },
    nom: {
      type: DataTypes.STRING,
      allowNull: false,
    },
    numero: {
      type: DataTypes.INTEGER,
      allowNull: false,
      unique: true,

    },
    password: {
      type: DataTypes.STRING,
      allowNull: false,
     
    },
    path: {
      type: DataTypes.STRING,
      allowNull: false,
    },
    latitude: {
      type: DataTypes.DOUBLE,
      allowNull: false,
    },
    longitude: {
      type: DataTypes.DOUBLE,
      allowNull: false,
    },
    etat: {
      type: DataTypes.STRING,
      allowNull: false,
    },
    email: {
      type: DataTypes.STRING,
      allowNull: false,
      unique:true,
      validate: {
        isEmail: {
          msg: 'Invalid email format',
        },
      },
    },
  }, {
    tableName: 'pharmacie' ,
    timestamps: false
  });

module.exports = pharmacie;

File: lib/Model/pharmacie.dart
import "package:flutter/material.dart";

class Pharmacie {
  int? id_pharmacie;
  String nom;
  int numero;
  String password;
  String? path;
  double latitude;
  double longitude;
  String email;
  String etat;

  Pharmacie({
    this.id_pharmacie,
    required this.nom,
    required this.numero,
    required this.password,
    this.path,
    required this.latitude,
    required this.longitude,
    required this.email,
    required this.etat,
  });

  factory Pharmacie.fromJson(Map<String, dynamic> json) {
    return Pharmacie(
      id_pharmacie: json['id_pharmacie'],
      nom: json['nom'],
      numero: json['numero'],
      password: json['password'],
      path: json['path'],
      latitude: json['latitude'],
      longitude: json['longitude'],
      email: json['email'],
      etat: json['etat'],
    );
  }

  Map<String, dynamic> toJson() {
    final date = <String, dynamic>{};
    date['id_pharmacie'] = id_pharmacie;
    date['latitude'] = latitude;
    date['longitude'] = longitude;
    date['nom'] = nom;
    date['numero'] = numero;
    date['password'] = password;
    date['email'] = email;
    return date;
  }
}
